platform/pulpcore/download/single.py

```diff
@@ -173,36 +201,31 @@ def _validate(self):
                 log.exception(unexpected)
                 raise unexpected
 
-    @delegate
     def _on_prepare(self):
         """
         Prepared to be executed.
         """
-        pass
+        self.router.send(Prepared(self))
 
-    @delegate
     def _on_reply(self):
         """
         A reply has been received.
         """
-        pass
+        self.router.send(Replied(self))
 
-    @delegate
     def _on_succeeded(self):
         """
         The download has succeeded.
         """
-        pass
+        self.router.send(Succeeded(self))
 
-    @delegate
     def _on_failed(self):
         """
         The download has failed.
         All efforts to repair and retry have failed.
         """
-        pass
+        self.router.send(Failed(self))
 
-    @delegate
     def _on_error(self, error):
         """
         The download has encountered an error.
```

Comment:
The gh interface wouldnt let me comment on right line, but the next line mentions a delegate.